lib/src/home.dart

```diff
@@ -1,88 +1,91 @@
-import 'package:bloc_sample/src/authentication.dart';
 import 'package:bloc_sample/src/home_bloc.dart';
 import 'package:bloc_sample/src/home_repository.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:shared_preferences/shared_preferences.dart';
+import 'package:bloc_sample/src/authentication_bloc.dart';
 
 @immutable
 class Home extends StatelessWidget {
   const Home({
     Key key,
   }) : super(key: key);
   @override
-  Widget build(BuildContext context) => RepositoryProvider<HomeRepository>(
-        create: (context) => HomeRepositoryImpl(
-          user: Authentication.of(context).user,
-          sharedPreferences: RepositoryProvider.of<SharedPreferences>(
-            context,
-            listen: true,
+  Widget build(BuildContext context) {
+    final user = context.select(
+      (AuthenticationBLoC bloc) => bloc.state
+          .maybeWhen(authenticated: (user) => user, orElse: () => null),
+    );
+    return RepositoryProvider<HomeRepository>(
+      create: (context) => HomeRepositoryImpl(
+        user: user,
+        sharedPreferences: context.read<SharedPreferences>(),
+      ),
+      child: BlocProvider<HomeBLoC>(
+        create: (context) => HomeBLoC(
+          repository: context.read<HomeRepository>(),
```

Comment:
If the repository changes (but as we see above, it cannot) - the BLoC will not be recreated